packages/babel-plugin-transform-es2015-modules-commonjs/src/index.js

```diff
@@ -126,289 +129,287 @@ export default function () {
     }
   };
 
-  return {
-    inherits: require("babel-plugin-transform-strict-mode"),
-
-    visitor: {
-      ThisExpression(path, state) {
-        // If other plugins run after this plugin's Program#exit handler, we allow them to
-        // insert top-level `this` values. This allows the AMD and UMD plugins to
-        // function properly.
-        if (this.ranCommonJS) return;
-
-        if (
-          state.opts.allowTopLevelThis !== true &&
-          !path.findParent((path) => !path.is("shadow") &&
-          THIS_BREAK_KEYS.indexOf(path.type) >= 0)
-        ) {
-          path.replaceWith(t.identifier("undefined"));
-        }
-      },
+  return Object.assign(
```

Comment:
This only works by coincidence because no keys clash, no actual merging is done.

Maybe it's ok in this case, but it's still... weird 😕